src/Builder/CliMenuBuilder.php

```diff
@@ -560,6 +572,18 @@ private function propagateStyles(CliMenu $menu, array $items = [])
         $currentItems = !empty($items) ? $items : $menu->getItems();
 
         foreach ($currentItems as $item) {
+            if ($item instanceof CheckboxItem
```

Comment:
The late style binding here really helps reduce complexity. As more item types are refactored to support item-level styling, they should be added here to make them work properly.